onedrivesdk/src/main/java/com/onedrive/sdk/http/BaseRequestBuilder.java

```diff
@@ -94,6 +96,11 @@ public String getRequestUrl() {
      * @return The base url for this request.
      */
     public String getRequestUrlWithAdditionalSegment(final String urlSegment) {
-        return mRequestUrl + "/" + urlSegment;
+        try {
+            return mRequestUrl + "/" + URLEncoder.encode(urlSegment, "UTF-8");
+        } catch (UnsupportedEncodingException ex) {
+            mClient.getLogger().logError("urlSegment '" + urlSegment + "' could not be encoded into UTF-8", ex);
+            return mRequestUrl + urlSegment;
```

Comment:
going to add missing "/", another question for @peternied is shall we throw the unsupported here or just return the url as it is. Throwing out exception means Template should also be changed as well and user should handle the invalid input.